rdbms/src/main/java/io/spine/server/storage/jdbc/query/ColumnAwareWriteQuery.java

```diff
@@ -41,6 +48,27 @@
         return columnTypeRegistry;
     }
 
+    /**
+     * Obtains the name-to-index transformer for the specified record.
+     *
+     * @param record           the record to extract column names
+     * @param firstColumnIndex the index of the first entity column in a query
+     * @return a {@code Function} transforming a column name to the its index
+     * @see io.spine.server.entity.storage.ColumnRecords#feedColumnsTo
+     */
+    Function<String, Integer> getTransformer(EntityRecordWithColumns record,
+                                             int firstColumnIndex) {
+        final Map<String, Integer> result = new HashMap<>();
+
+        int index = firstColumnIndex;
+        for (String entry : record.getColumnNames()) {
```

Comment:
As of now, this list isn't sorted. So the column numbers are going to be random each time this code is executed.

Reply:
There is test, `RecordStorageShould.update_entity_column_values` in `core-java`, which tests such cases.